### PR TITLE
replace rust_input_handle_t with InputHandleWrapper (without Read)

### PR DESCRIPTION
**Pull request by burrbull**
cc @Mrmaxmeier

### CODE REVIEW

File: dpx/src/dpx_cidtype0.rs [hunk 1]
```diff
@@ -2051,10 +2016,11 @@ pub unsafe extern "C" fn CIDFont_type0_t1dofont(mut font: *mut CIDFont) {
         pdf_ref_obj((*font).descriptor),
     );
     let handle = dpx_open_type1_file((*font).ident);
-    if handle.is_null() {
+    if handle.is_none() {
```
Comment (mulimoen) [inline dpx/src/dpx_cidtype0.rs:2019]: You can use `expect()` here

File: dpx/src/dpx_epdf.rs [hunk 2]
```diff
@@ -30,36 +30,36 @@
 
 use crate::warn;
 
-use super::dpx_mem::xmalloc;
-use super::dpx_pdfdoc::pdf_doc_add_page_content;
+//use super::dpx_mem::xmalloc;
```
Comment (mulimoen) [inline dpx/src/dpx_epdf.rs:33]: Remove instead of commenting out

Comment (burrbull) [reply dpx/src/dpx_epdf.rs:33]: #145 need decision

File: dpx/src/dpx_cidtype0.rs (hunk 1 above)
Comment (burrbull) [reply dpx/src/dpx_cidtype0.rs:2019]: We have many places where `panic!` can be replaced with `assert!` (expect). I did not try to do this here to simplify review.
It is easier to find all panics and handle.

**Comment by Mrmaxmeier** [comment]
This introduces a bunch of `error: serious internal bug: unexpected handle in input close: 0x...`

```
total regressed (reg): 53
total regressed (fix): 665
    fixed regressions: 0
      new regressions: 612
changed but not fixed: 16
```
https://tt.ente.ninja/#/compare/oxidize-v0.1.11-725-g83fc2fa8/input-v0.1.11-736-g2c01386d

**Comment by Mrmaxmeier** [comment]
With dbc13b1:
```
total regressed (reg): 53
total regressed (fix): 54
    fixed regressions: 0
      new regressions: 1
changed but not fixed: 0
```
https://tt.ente.ninja/#/compare/oxidize-v0.1.11-725-g83fc2fa8/input-v0.1.11-737-gdbc13b16

**Comment by burrbull** [comment]
Interesting situation.
I compiled tectonic from `oxidize` branch in release mode and ran:
```
~/builds/tectonic/release/tectonic '/home/burrbull/1/Untitled_Folder/pccp_2017_cu3cocn6.tex'
note: this is a BETA release; ask questions and report bugs at https://tectonic.newton.cx/
Running TeX ...
note: downloading SHA256SUM

...

Interesting note: downloading pcrr8r.tfm
note: downloading ucrr8a.pfb
note: downloading zptmcm7v.vf
Writing /home/burrbull/1/Untitled_Folder/pccp_2017_cu3cocn6.pdf (1576143 bytes)
Skipped writing 2 intermediate files (use --keep-intermediates to keep them)
```
Delete only `pccp_2017_cu3cocn6.pdf` and rerun:
```
➜  ~ ~/builds/tectonic/release/tectonic '/home/burrbull/1/Untitled_Folder/pccp_2017_cu3cocn6.tex'
note: this is a BETA release; ask questions and report bugs at https://tectonic.newton.cx/
Running TeX ...
note: warnings were issued by the TeX engine; use --print and/or --keep-logs for details.
Running BibTeX ...
warning: errors were issued by BibTeX, but were ignored; use --print and/or --keep-logs for details.
Rerunning TeX because bibtex was run ...
Rerunning TeX because "pccp_2017_cu3cocn6.aux" changed ...
Rerunning TeX because "pccp_2017_cu3cocn6.aux" changed ...
Running xdvipdfmx ...
[1]    1803 segmentation fault (core dumped)  ~/builds/tectonic/release/tectonic 
```

**Comment by burrbull** [comment]
Looks like the problem is not in this PR, so let's merge it.

**Pull request merged**